Public/Src/Cache/MemoizationStore/Library/Stores/RocksDbMemoizationStore.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+extern alias Async;
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.Threading;
+using System.Threading.Tasks;
+using BuildXL.Cache.ContentStore.Distributed.NuCache;
+using BuildXL.Cache.ContentStore.Interfaces.Extensions;
+using BuildXL.Cache.ContentStore.Interfaces.Logging;
+using BuildXL.Cache.ContentStore.Interfaces.Results;
+using BuildXL.Cache.ContentStore.Interfaces.Sessions;
+using BuildXL.Cache.ContentStore.Interfaces.Time;
+using BuildXL.Cache.ContentStore.Interfaces.Tracing;
+using BuildXL.Cache.ContentStore.Tracing.Internal;
+using BuildXL.Cache.MemoizationStore.Interfaces.Results;
+using BuildXL.Cache.MemoizationStore.Interfaces.Sessions;
+using BuildXL.Cache.MemoizationStore.Interfaces.Stores;
+using BuildXL.Cache.MemoizationStore.Sessions;
+using BuildXL.Cache.MemoizationStore.Tracing;
+
+namespace BuildXL.Cache.MemoizationStore.Stores
+{
+    /// <summary>
+    ///     An IMemoizationStore implementation using RocksDb.
+    /// </summary>
+    public class RocksDbMemoizationStore : IMemoizationStore
+    {
+        private const string Component = nameof(RocksDbMemoizationStore);
+
+        private bool _disposed;
+        private IClock _clock;
+        private RocksDbContentLocationDatabaseConfiguration _config;
+        private RocksDbContentLocationDatabase _cldb;
+
+        /// <summary>
+        ///     Store tracer.
+        /// </summary>
+        private readonly RocksDbMemoizationStoreTracer Tracer;
+
+        /// <inheritdoc />
+        public bool StartupStarted { get; private set; }
+
+        /// <inheritdoc />
+        public bool StartupCompleted { get; private set; }
+
+        /// <inheritdoc />
+        public bool ShutdownStarted { get; private set; }
+
+        /// <inheritdoc />
+        public bool ShutdownCompleted { get; private set; }
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="RocksDbMemoizationStore"/> class.
+        /// </summary>
+        public RocksDbMemoizationStore(ILogger logger, IClock clock, RocksDbContentLocationDatabaseConfiguration config)
+        {
+            Contract.Requires(logger != null);
+            Contract.Requires(config != null);
+            Contract.Requires(clock != null);
+
+            Tracer = new RocksDbMemoizationStoreTracer(logger, Component);
+            _clock = clock;
+            _config = config;
+            _cldb = new RocksDbContentLocationDatabase(clock, config, () => new MachineId[] { });

```

Comment:
_rocksDb? or _database?

Reply:
_cldb looks a bit weird, imo.

---
In reply to: [305162346](https://github.com/microsoft/BuildXL/pull/621#discussion_r305162346) [](ancestors = 305162346)